### ISSUE TITLE
[Optimization] Calculate the lookAt matrix only when the camera matrix changed.

### ISSUE THREAD

**Issue by malortie** [Solokiller/HL_Tools#136]
The lookAt matrix (view matrix) appears to be calculated each time the scene is drawn.

https://github.com/Solokiller/HL_Tools/blob/597a3a3bdd23daa5eaf2cf32d5c86a5a16cbf16a/src/tools/hlam/graphics/Scene.cpp#L357

https://github.com/Solokiller/HL_Tools/blob/597a3a3bdd23daa5eaf2cf32d5c86a5a16cbf16a/src/tools/hlam/graphics/Scene.cpp#L321-L328

The recommendation would be to store the lookAt matrix as a variable that would be updated when the camera matrix (`_viewMatrix`) is updated.

https://github.com/Solokiller/HL_Tools/blob/597a3a3bdd23daa5eaf2cf32d5c86a5a16cbf16a/src/tools/hlam/graphics/Camera.cpp#L9-L20

The lookAt matrix variable could then be used as follows:

```cpp
glLoadMatrix(glm::value_ptr(camera->GetLookAtMatrix()));
```

In addition, it is recommended to rename the current `_viewMatrix` to `_cameraMatrix` and use `_viewMatrix` to store the lookAt matrix instead.

**Comment by Solokiller** [comment]
I've renamed the view matrix to model matrix instead in keeping with the model-view-projection matrix naming convention.

**Comment by Solokiller** [comment]
I've provided an alpha build that fixes this issue. Can you test it and see if it's fixed?

https://github.com/Solokiller/HL_Tools/releases/tag/HLAM-V1.1.0-alpha001

**Comment by malortie** [comment]
I tested with the 1.1.0 alpha 002. It should also work in the alpha 001, as the only difference between alpha 002 and 001 is the [fix for the flat shade](https://github.com/Solokiller/HL_Tools/compare/HLAM-V1.1.0-alpha001...HLAM-V1.1.0-alpha002).

I tested all camera modes and they appear to work correctly.

I also looked at the new code and it should to be ok, given the view matrix is calculated with the same input parameters and is updated each time the model matrix changes.
